src/common/tree_renderer/yaml_tree_renderer.cpp

```diff
@@ -0,0 +1,144 @@
+#include "duckdb/common/tree_renderer/yaml_tree_renderer.hpp"
+
+#include "duckdb/common/string_util.hpp"
+#include "duckdb/common/typedefs.hpp"
+#include "duckdb/execution/physical_operator.hpp"
+#include "duckdb/parallel/pipeline.hpp"
+#include "duckdb/planner/logical_operator.hpp"
+#include "fmt/printf.h"
+
+#include <ostream>
+#include <sstream>
+
+namespace duckdb {
+
+string YAMLTreeRenderer::ToString(const LogicalOperator &op) {
+	std::stringstream ss;
+	Render(op, ss);
+	return ss.str();
+}
+
+string YAMLTreeRenderer::ToString(const PhysicalOperator &op) {
+	std::stringstream ss;
+	Render(op, ss);
+	return ss.str();
+}
+
+string YAMLTreeRenderer::ToString(const ProfilingNode &op) {
+	std::stringstream ss;
+	Render(op, ss);
+	return ss.str();
+}
+
+string YAMLTreeRenderer::ToString(const Pipeline &op) {
+	std::stringstream ss;
+	Render(op, ss);
+	return ss.str();
+}
+
+void YAMLTreeRenderer::Render(const LogicalOperator &op, std::ostream &ss) {
+	auto tree = RenderTree::CreateRenderTree(op);
+	ToStream(*tree, ss);
+}
+
+void YAMLTreeRenderer::Render(const PhysicalOperator &op, std::ostream &ss) {
+	auto tree = RenderTree::CreateRenderTree(op);
+	ToStream(*tree, ss);
+}
+
+void YAMLTreeRenderer::Render(const ProfilingNode &op, std::ostream &ss) {
+	auto tree = RenderTree::CreateRenderTree(op);
+	ToStream(*tree, ss);
+}
+
+void YAMLTreeRenderer::Render(const Pipeline &op, std::ostream &ss) {
+	auto tree = RenderTree::CreateRenderTree(op);
+	ToStream(*tree, ss);
+}
+
+void YAMLTreeRenderer::ToStreamInternal(RenderTree &root, std::ostream &ss) {
+	RenderRecursive(root, ss, 0, 0, 0);
+}
+
+struct EscapedString {
+	explicit EscapedString(const string &str) : str(str) {};
+	const string &str;
+};
+
+std::ostream &operator<<(std::ostream &out, const EscapedString &es) {
+	out << '"';
```

Comment:
This is rather over-eager with adding quotes which reduces readability - can we only add quotes when required (i.e. when there are special characters in the string)?

Reply:
I copy this from postgres. As comment there, 

> the YAML quoting rules are ridiculously complicated... so we chose to just quote everything

https://github.com/postgres/postgres/blob/84914e964b447a63c5208ca2ec11ec3abfda0879/src/backend/commands/explain_format.c?plain=1#L700-L714